### PR TITLE
Working test to check that dockerfile generation is working

### PR DESCRIPTION
**Pull request by JulianGindi**

### CODE REVIEW

**Comment by JulianGindi** [comment]
I will probably make the "compare files" code generic so we can use it to compare any two generated files

**Comment by CorneliusIV** [comment]
:+1:

**Comment by oudeismetis** [comment]
Other than the milagro bit, this is fan-tastic. :sparkles:

File: build-api/test/sample_data.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "deploy_key": "03d77f0d623f0491d6ff5327bb992cbfc9441a0b",
```
Comment (CorneliusIV) [inline build-api/test/sample_data.json:2]: Would we break anything if we replace this with a dummy deploy key

Comment (oudeismetis) [inline build-api/test/sample_data.json:2]: It's already a dummy deploy key. Github detected this commit and expired this key.

Comment (CorneliusIV) [inline build-api/test/sample_data.json:2]: ah right, perfect

**Comment by CorneliusIV** [comment]
:+1:

File: build-api/api.go [hunk 2]
```diff
@@ -1,73 +1,118 @@
...
+func BuildDockerFile(p martini.Params, r render.Render, dockerInfo DockerInfo) {
+	err := generateDockerFile(dockerInfo, ".")
+
+	if err != nil {
+		r.JSON(500, map[string]interface{}{"success": false})
+	}
+
+	logging.LogToFile("Dockerfile generated successfully...building container...")
+	// TODO: Obviously change this
+	go build("/Users/gindi/Desktop/tmp_build_dir", dockerInfo.REPO_NAME)
```
Comment (JulianGindi) [inline build-api/api.go]: Going to make this a var

File: build-api/api.go [hunk 3]
```diff
@@ -1,73 +1,118 @@
...
+	com <- randomTag
 }
 
 func main() {
 	m := martini.Classic()
 	m.Use(render.Renderer())
-	m.Get("/", func() string {
+
+	// Simple 'health' endpoint for AWS load-balancer health checks
+	m.Get("/health", func() string {
```
Comment (CorneliusIV) [inline build-api/api.go:45]: Nice, I was thinking we should remove this but this is a good idea.

File: build-api/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+RUN eval `ssh-agent -s` && \
+    ssh-add ~/.ssh/id_rsa && \
+    ssh-keyscan github.com >> /root/.ssh/known_hosts
+
+RUN git clone -b master git@github.com:istrategylabs/franklin-test.git
+RUN cd franklin-test && git checkout 7b3af09db495427d067704b5e9290d8b81d682c5 
+
+
+RUN cd franklin-test && npm install
+RUN cd franklin-test && npm run build
```
Comment (CorneliusIV) [inline build-api/Dockerfile]: Did you mean to commit this? We probably might want to add this to the .gitignore

File: build-api/api.go [hunk 5]
```diff
@@ -1,73 +1,118 @@
...
-	tearDown := exec.Command("scripts/tear_down_project.sh")
-	if err := tearDown.Run(); err != nil {
-		fmt.Println(os.Stderr, err)
-	}
+// buildDockerContainer executes a docker build command and assigns it a random tag
+func buildDockerContainer(com chan string) {
+	// First we will seed the random number generator
+	rand.Seed(time.Now().UnixNano())
+	randomTag := strconv.Itoa(rand.Intn(1000))
+	exec.Command("docker", "build", "--no-cache=True", "-t", randomTag, ".").Run()
```
Comment (CorneliusIV) [inline build-api/api.go:34]: Should we log anything from here?

File: build-api/Dockerfile (hunk 4 above)
Comment (JulianGindi) [inline build-api/Dockerfile]: Yup, good call

File: build-api/api.go (hunk 5 above)
Comment (JulianGindi) [inline build-api/api.go:34]: Probably, I'll look what I can do

**Comment by JulianGindi** [comment]
Also, don't worry, history will be cleaned up before rebase.

**Pull request closed without merging**